src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { CaseField } from '../../../domain';
+
+@Pipe({
+  name: 'ccdCYAPageLabelFilter'
+})
+export class CcdCYAPageLabelFilterPipe implements PipeTransform {
+
+  public transform(caseFields: CaseField[]): CaseField[] {
+    return caseFields.map((caseField: CaseField) => {
+
+      if (caseField.field_type.complex_fields && caseField.field_type.complex_fields.length) {
+        caseField.field_type.complex_fields = this.removeLabels(caseField.field_type.complex_fields);
+      }
+
+      return caseField;
+    });
+  }
+
+  private removeLabels = (complexFields: CaseField[]): CaseField[] =>
```

Comment:
Should this method be called removeLabels? It's not removing anything rather applying a filter. I

Reply:
Well it's taking a list of `CaseFields[]` and removing all CaseFields of type Label.

Reply:
I understand that. But, the private method removeLabels is not removing it. Rather the result of the method removeLabels is doing that. So, the method could be called getNonLabelComplexFieldType? Agree?

Reply:
Oh I see what you mean. What you're saying makes sense I will change the method name. Agreed